c7n/resources/rds.py

```diff
@@ -110,6 +112,46 @@ def get_resources(self, resource_ids):
                     DBInstanceIdentifier=db_id)['DBInstances'])
         return results
 
+    def add_tags(self, client, dbs):
+        """
+        Adds the tags to the list of databases
+        """
+        account_id = self.get_account_id()
+        fn = partial(self.process_tags, account_id=account_id ,client=client)
+        with self.executor_factory(max_workers=3) as w:
+            return list(w.map(fn,dbs))
+
+    def process_tags(self, db, **kwargs):
+        region = self.get_region(db)
+        name = db['DBInstanceIdentifier']
+        arn = "arn:aws:rds:%s:%s:db:%s" % (region, kwargs['account_id'], name)
+        tag_list = (
+            kwargs['client'].list_tags_for_resource(ResourceName=arn)['TagList']
+        )
+        tags = []
+        if tag_list:
+            tags.extend(tag_list)
+        db['Tags'] = tags
+        return db
+
+    def get_account_id(self):
+        # TODO: This just gets a security group in the account
+        # and uses that as the ARN of the account. See notes
+        # at top of file for how else we can do this
+        client = self.session_factory().client('ec2')
```

Comment:
I didn't love the security groups, but like you said with IAM and SAML there are issues depending on how you're running. SGs seemed like the lowest common denominator.

Will iam.list_roles work for both cases?